daemon/graphdriver/overlay2/overlay.go

```diff
@@ -213,7 +214,12 @@ func parseOptions(options []string) (*overlayOptions, error) {
 			if err != nil {
 				return nil, err
 			}
-
+		case "overlay2.size":
+			size, err := units.RAMInBytes(val)
+			if err != nil {
+				return nil, err
+			}
+			o.quota.Size = uint64(size)
```

Comment:
IOW, in Init() this check should be there and driver initialization should fail if quota can't be supported and user asked for it.